### PR TITLE
Add support for categories

### PR DESCRIPTION
**Pull request by paprikati**
Introduced a new property `categories` on `resource_catalog_type` which you can use to make your catalog easier to navigate.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ build:
 # Clients
 ################################################################################
 
-.PHONY: internal/client
+.PHONY: internal/client/client.gen.go
```
Comment (paprikati) [inline Makefile:29]: unrelated: without this, you have to manually delete the client before re-generating it

File: internal/apischema/openapi3.json [hunk 2]
```diff
@@ -6049,56 +6079,6 @@
         }
       }
     },
-    "/v2/managed_resources": {
```
Comment (pipt) [inline internal/apischema/openapi3.json:6052]: Is this removal intentional?

**Review by pipt — APPROVED** [review]
One question about the managed resources API removal from the API json, but looks good :+1:

File: internal/apischema/openapi3.json (hunk 2 above)
Comment (paprikati) [reply internal/apischema/openapi3.json:6052]: Yes, I've copied over all the schemas which is what lojo was expecting when he made the change. We only read 'secret' but to make it harder to accidentally pull in the wrong schema we now have all of them in a single folder
